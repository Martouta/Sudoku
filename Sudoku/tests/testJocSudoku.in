3

3

4
medio

4
trivi

3

2

1

1

1

0


Explicación:

Escribe la n del sudoku
3 //introducimos el tamaño del sudoku (en este caso, como la n es 9 el tablero del sudoku será de 9x9)

3 //obtenemos la dificultad inicial del sudoku asignada al crearlo (dificil en este caso)

4 //modificamos la dificultad para probar que funciona correctamente
Introduce la dificultad del sudoku manualmente
trivial, facil, medio, dificil
medio

4 //probamos de introducir un valor de dificultad no válido para comprobar que sigue funcionando correctamente
Introduce la dificultad del sudoku manualmente
trivial, facil, medio, dificil
trivi

3 //obtenemos la dificultad otra vez para comprobar que el caso anterior no nos ha modificado la dificultad erróneamente

2 //obtenemos el id del juego

1 //obtenemos varias pistas para comprobar el correcto funcionamiento del método

1

1

0