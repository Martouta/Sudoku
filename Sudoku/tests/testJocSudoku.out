Se ha insertado el valor 1 en la fila 2 y columna 5
Se ha ejecutado el algoritmo, 327 iteraciones
Se ha insertado el valor 4 en la fila 1 y columna 2
Se ha ejecutado el algoritmo, 252 iteraciones
Se ha insertado el valor 6 en la fila 1 y columna 8
Se ha ejecutado el algoritmo, 172 iteraciones
Se ha insertado el valor 2 en la fila 2 y columna 7
Se ha ejecutado el algoritmo, 166 iteraciones
Se ha insertado el valor 2 en la fila 8 y columna 5
Se ha ejecutado el algoritmo, 169 iteraciones
Se ha insertado el valor 1 en la fila 3 y columna 3
Se ha ejecutado el algoritmo, 203 iteraciones
Se ha insertado el valor 3 en la fila 4 y columna 1
Se ha ejecutado el algoritmo, 227 iteraciones
Se ha insertado el valor 9 en la fila 1 y columna 1
Se ha ejecutado el algoritmo, 245 iteraciones
Se ha insertado el valor 9 en la fila 4 y columna 7
Se ha ejecutado el algoritmo, 250 iteraciones
Se ha insertado el valor 7 en la fila 8 y columna 6
Se ha ejecutado el algoritmo, 161 iteraciones
Se ha insertado el valor 5 en la fila 4 y columna 0
Se ha ejecutado el algoritmo, 176 iteraciones
Se ha insertado el valor 8 en la fila 1 y columna 0
Se ha ejecutado el algoritmo, 137 iteraciones
Se ha insertado el valor 4 en la fila 0 y columna 6
Se ha ejecutado el algoritmo, 270 iteraciones
Se ha insertado el valor 3 en la fila 8 y columna 4
Se ha ejecutado el algoritmo, 466 iteraciones
Se ha insertado el valor 9 en la fila 7 y columna 4
Se ha ejecutado el algoritmo, 277 iteraciones
Se ha insertado el valor 4 en la fila 3 y columna 1
Se ha ejecutado el algoritmo, 276 iteraciones
Se ha insertado el valor 6 en la fila 5 y columna 5
Se ha ejecutado el algoritmo, 272 iteraciones
Se ha insertado el valor 5 en la fila 7 y columna 6
Se ha ejecutado el algoritmo, 364 iteraciones
Se ha insertado el valor 7 en la fila 1 y columna 5
Se ha ejecutado el algoritmo, 17373 iteraciones
Se ha insertado el valor 2 en la fila 3 y columna 4
Se ha ejecutado el algoritmo, 726154 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 2 en la fila 4 y columna 8
Se ha ejecutado el algoritmo, 1560416 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 3 en la fila 5 y columna 3
Se ha ejecutado el algoritmo, 24570 iteraciones
Se ha insertado el valor 7 en la fila 2 y columna 2
Se ha ejecutado el algoritmo, 6566 iteraciones
Se ha insertado el valor 8 en la fila 8 y columna 2
Se ha ejecutado el algoritmo, 3995 iteraciones
Se ha insertado el valor 8 en la fila 3 y columna 5
Se ha ejecutado el algoritmo, 0 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 2 en la fila 6 y columna 8
Se ha ejecutado el algoritmo, 3517 iteraciones
Se ha insertado el valor 5 en la fila 0 y columna 7
Se ha ejecutado el algoritmo, 25720 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 6 en la fila 3 y columna 2
Se ha ejecutado el algoritmo, 1360 iteraciones
Se ha insertado el valor 2 en la fila 5 y columna 0
Se ha ejecutado el algoritmo, 385 iteraciones
Se ha insertado el valor 2 en la fila 0 y columna 4
Se ha ejecutado el algoritmo, 0 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 9 en la fila 3 y columna 5
Se ha ejecutado el algoritmo, 289 iteraciones
Se ha insertado el valor 8 en la fila 4 y columna 5
Se ha ejecutado el algoritmo, 233 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 9 en la fila 2 y columna 8
Se ha ejecutado el algoritmo, 1510 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 6 en la fila 0 y columna 0
Se ha ejecutado el algoritmo, 612 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 8 en la fila 5 y columna 8
Se ha ejecutado el algoritmo, 0 iteraciones
Sudoku no resoluble, borrando última casilla
Se ha insertado el valor 6 en la fila 8 y columna 3
Se ha ejecutado el algoritmo, 3991 iteraciones
Se ha ejecutado el algoritmo, 3991 iteraciones
Se ha insertado el valor 1 en la fila 0 y columna 0
Se ha insertado el valor 2 en la fila 0 y columna 1
Se ha insertado el valor 5 en la fila 0 y columna 2
Se ha insertado el valor 8 en la fila 0 y columna 3
Se ha insertado el valor 6 en la fila 0 y columna 4
Se ha insertado el valor 3 en la fila 0 y columna 5
Se ha insertado el valor 4 en la fila 0 y columna 6
Se ha insertado el valor 7 en la fila 0 y columna 7
Se ha insertado el valor 9 en la fila 0 y columna 8
Se ha insertado el valor 8 en la fila 1 y columna 0
Se ha insertado el valor 9 en la fila 1 y columna 1
Se ha insertado el valor 4 en la fila 1 y columna 2
Se ha insertado el valor 5 en la fila 1 y columna 3
Se ha insertado el valor 2 en la fila 1 y columna 4
Se ha insertado el valor 7 en la fila 1 y columna 5
Se ha insertado el valor 3 en la fila 1 y columna 6
Se ha insertado el valor 1 en la fila 1 y columna 7
Se ha insertado el valor 6 en la fila 1 y columna 8
Se ha insertado el valor 3 en la fila 2 y columna 0
Se ha insertado el valor 6 en la fila 2 y columna 1
Se ha insertado el valor 7 en la fila 2 y columna 2
Se ha insertado el valor 9 en la fila 2 y columna 3
Se ha insertado el valor 4 en la fila 2 y columna 4
Se ha insertado el valor 1 en la fila 2 y columna 5
Se ha insertado el valor 8 en la fila 2 y columna 6
Se ha insertado el valor 2 en la fila 2 y columna 7
Se ha insertado el valor 5 en la fila 2 y columna 8
Se ha insertado el valor 7 en la fila 3 y columna 0
Se ha insertado el valor 4 en la fila 3 y columna 1
Se ha insertado el valor 6 en la fila 3 y columna 2
Se ha insertado el valor 1 en la fila 3 y columna 3
Se ha insertado el valor 5 en la fila 3 y columna 4
Se ha insertado el valor 9 en la fila 3 y columna 5
Se ha insertado el valor 2 en la fila 3 y columna 6
Se ha insertado el valor 3 en la fila 3 y columna 7
Se ha insertado el valor 8 en la fila 3 y columna 8
Se ha insertado el valor 5 en la fila 4 y columna 0
Se ha insertado el valor 3 en la fila 4 y columna 1
Se ha insertado el valor 1 en la fila 4 y columna 2
Se ha insertado el valor 2 en la fila 4 y columna 3
Se ha insertado el valor 8 en la fila 4 y columna 4
Se ha insertado el valor 4 en la fila 4 y columna 5
Se ha insertado el valor 6 en la fila 4 y columna 6
Se ha insertado el valor 9 en la fila 4 y columna 7
Se ha insertado el valor 7 en la fila 4 y columna 8
Se ha insertado el valor 2 en la fila 5 y columna 0
Se ha insertado el valor 8 en la fila 5 y columna 1
Se ha insertado el valor 9 en la fila 5 y columna 2
Se ha insertado el valor 3 en la fila 5 y columna 3
Se ha insertado el valor 7 en la fila 5 y columna 4
Se ha insertado el valor 6 en la fila 5 y columna 5
Se ha insertado el valor 1 en la fila 5 y columna 6
Se ha insertado el valor 5 en la fila 5 y columna 7
Se ha insertado el valor 4 en la fila 5 y columna 8
Se ha insertado el valor 6 en la fila 6 y columna 0
Se ha insertado el valor 7 en la fila 6 y columna 1
Se ha insertado el valor 3 en la fila 6 y columna 2
Se ha insertado el valor 4 en la fila 6 y columna 3
Se ha insertado el valor 1 en la fila 6 y columna 4
Se ha insertado el valor 5 en la fila 6 y columna 5
Se ha insertado el valor 9 en la fila 6 y columna 6
Se ha insertado el valor 8 en la fila 6 y columna 7
Se ha insertado el valor 2 en la fila 6 y columna 8
Se ha insertado el valor 4 en la fila 7 y columna 0
Se ha insertado el valor 1 en la fila 7 y columna 1
Se ha insertado el valor 2 en la fila 7 y columna 2
Se ha insertado el valor 7 en la fila 7 y columna 3
Se ha insertado el valor 9 en la fila 7 y columna 4
Se ha insertado el valor 8 en la fila 7 y columna 5
Se ha insertado el valor 5 en la fila 7 y columna 6
Se ha insertado el valor 6 en la fila 7 y columna 7
Se ha insertado el valor 3 en la fila 7 y columna 8
Se ha insertado el valor 9 en la fila 8 y columna 0
Se ha insertado el valor 5 en la fila 8 y columna 1
Se ha insertado el valor 8 en la fila 8 y columna 2
Se ha insertado el valor 6 en la fila 8 y columna 3
Se ha insertado el valor 3 en la fila 8 y columna 4
Se ha insertado el valor 2 en la fila 8 y columna 5
Se ha insertado el valor 7 en la fila 8 y columna 6
Se ha insertado el valor 4 en la fila 8 y columna 7
Se ha insertado el valor 1 en la fila 8 y columna 8
Se ha ejecutado el algoritmo, 6053 iteraciones
++--------++--------++--------++
|| 0| 0| 0|| 0| 0| 0|| 4| 0| 0||
|| 8| 9| 4|| 0| 0| 7|| 0| 0| 6||
|| 0| 0| 7|| 0| 0| 1|| 0| 2| 0||
++--------++--------++--------++
|| 0| 4| 6|| 1| 0| 9|| 0| 0| 0||
|| 5| 3| 0|| 0| 0| 0|| 0| 9| 0||
|| 2| 0| 0|| 3| 0| 6|| 0| 0| 0||
++--------++--------++--------++
|| 0| 0| 0|| 0| 0| 0|| 0| 0| 2||
|| 0| 0| 0|| 0| 9| 0|| 5| 0| 0||
|| 0| 0| 8|| 6| 3| 2|| 7| 0| 0||
++--------++--------++--------++
++--------++--------++--------++
|| 1| 2| 5|| 8| 6| 3|| 4| 7| 9||
|| 8| 9| 4|| 5| 2| 7|| 3| 1| 6||
|| 3| 6| 7|| 9| 4| 1|| 8| 2| 5||
++--------++--------++--------++
|| 7| 4| 6|| 1| 5| 9|| 2| 3| 8||
|| 5| 3| 1|| 2| 8| 4|| 6| 9| 7||
|| 2| 8| 9|| 3| 7| 6|| 1| 5| 4||
++--------++--------++--------++
|| 6| 7| 3|| 4| 1| 5|| 9| 8| 2||
|| 4| 1| 2|| 7| 9| 8|| 5| 6| 3||
|| 9| 5| 8|| 6| 3| 2|| 7| 4| 1||
++--------++--------++--------++

dificil

medio

La dificultad introducida no es valida

medio

prova1

La posicion esta fuera del rango permitido
Pista en la posicion 0 0
++--------++--------++--------++
|| 1| 0| 0|| 0| 0| 0|| 4| 0| 0||
|| 8| 9| 4|| 0| 0| 7|| 0| 0| 6||
|| 0| 0| 7|| 0| 0| 1|| 0| 2| 0||
++--------++--------++--------++
|| 0| 4| 6|| 1| 0| 9|| 0| 0| 0||
|| 5| 3| 0|| 0| 0| 0|| 0| 9| 0||
|| 2| 0| 0|| 3| 0| 6|| 0| 0| 0||
++--------++--------++--------++
|| 0| 0| 0|| 0| 0| 0|| 0| 0| 2||
|| 0| 0| 0|| 0| 9| 0|| 5| 0| 0||
|| 0| 0| 8|| 6| 3| 2|| 7| 0| 0||
++--------++--------++--------++

1
La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
La posicion esta fuera del rango permitido
Pista en la posicion 2 0
++--------++--------++--------++
|| 1| 0| 0|| 0| 0| 0|| 4| 0| 0||
|| 8| 9| 4|| 0| 0| 7|| 0| 0| 6||
|| 3| 0| 7|| 0| 0| 1|| 0| 2| 0||
++--------++--------++--------++
|| 0| 4| 6|| 1| 0| 9|| 0| 0| 0||
|| 5| 3| 0|| 0| 0| 0|| 0| 9| 0||
|| 2| 0| 0|| 3| 0| 6|| 0| 0| 0||
++--------++--------++--------++
|| 0| 0| 0|| 0| 0| 0|| 0| 0| 2||
|| 0| 0| 0|| 0| 9| 0|| 5| 0| 0||
|| 0| 0| 8|| 6| 3| 2|| 7| 0| 0||
++--------++--------++--------++
Driver de Joc Sudoku
1: Pista
2: Get id del JocSudoku
3: Get dificultad del JocSudoku
4: Set la dificultad del JocSudoku
0: Salir
1
Pista en la posicion 5 6
++--------++--------++--------++
|| 1| 0| 0|| 0| 0| 0|| 4| 0| 0||
|| 8| 9| 4|| 0| 0| 7|| 0| 0| 6||
|| 3| 0| 7|| 0| 0| 1|| 0| 2| 0||
++--------++--------++--------++
|| 0| 4| 6|| 1| 0| 9|| 0| 0| 0||
|| 5| 3| 0|| 0| 0| 0|| 0| 9| 0||
|| 2| 0| 0|| 3| 0| 6|| 1| 0| 0||
++--------++--------++--------++
|| 0| 0| 0|| 0| 0| 0|| 0| 0| 2||
|| 0| 0| 0|| 0| 9| 0|| 5| 0| 0||
|| 0| 0| 8|| 6| 3| 2|| 7| 0| 0||
++--------++--------++--------++